x-pack/plugins/encrypted_saved_objects/server/routes/key_rotation.ts

```diff
@@ -40,6 +40,8 @@ export function defineKeyRotationRoutes({
       },
       options: {
         tags: ['access:rotateEncryptionKey'],
+        access: 'public',
```

Comment:
It's already publicly documented in https://www.elastic.co/guide/en/kibana/current/saved-objects-api-rotate-encryption-key.html. Does the "technical preview" state enables us to remove it from all docs?

Reply:
Since we publicly document this endpoint, I think it's fine to make it `public`. The JSDoc comment fro `public` definition is confusing though, I don't think `stable` is the requirement.

But at the same time this API should stay internal for Serverless, and Serverless docs shouldn'y mention this API (if we happen to have different docs for different offerings).

@jloleysens what would the appropriate way to deal with such API when it comes to OAS?

Reply:
Gonna repost what I think is best for SO APIs as this probably also applies here:

>What I think we could do instead: add a check for isServerless and only set SO CRUD APIs to public if not on serverless. That way we would be in line with the existing stack docs for onprem but diverge for serverless. This would be best done as a separate PR IMO as we'd also want to add some tests.

What do you think?

Reply:
> I don't think stable is the requirement.

Yeah, I agree. Though, I'm not certain whether this is best done by adding a new access level (`internal` | `public` | `unstable` | `system` etc) or just a separate piece of metadata that captures this. I'll create an issue for Core team to look at this.

Reply:
>> What I think we could do instead: add a check for isServerless and only set SO CRUD APIs to public if not on serverless. That way we would be in line with the existing stack docs for onprem but diverge for serverless. This would be best done as a separate PR IMO as we'd also want to add some tests.

> What do you think?

Yeah, sounds reasonable to me 👍

Reply:
I've removed the "access" declarations via ea522526147da85e8fe9f66dfebd3b4df01302d1 so this PR now just adds the descriptions.